scipy/_lib/decorator.py

```diff
@@ -390,7 +390,7 @@ def _dispatch(dispatch_args, *args, **kw):
             return func(*args, **kw)
 
         return FunctionMaker.create(
-            func, 'return _f_(%s, %%(shortsignature)s)' % dispatch_str,
+            func, f'return _f_({dispatch_str}, %%(shortsignature)s)',
             dict(_f_=_dispatch), register=register, default=func,
```

Comment:
I didn't fully understand the second argument in the `create()` call. Especially the part with `%%(shortsignature)s` - I wasn't sure if that was supposed to be a substitution or not.
In the file on main, I see there's only variable to substitute i.e. `dispatch_str`, so I think this change is OK. Also, the CI build was successful, so it seems to be. But i don't know how the test coverage is, so I just wanted to draw attention to this specifically.

Reply:
Shall I leave this as it is, then?

Reply:
gentle ping @ev-br, this is pretty much ready to go otherwise.

Reply:
I don't have an opinion, sorry. My git blame contribution was updatinging this vendored module (https://pypi.org/project/decorator/#description) to then-current version, probably in response to an issue which was relevant 9 years ago, but I have no recollection of. 

We can equally well either sync with the current pypi version or keep adding fixes that look reasonable, I suppose :-)

Reply:
My guess is that this is a simple typo, but let's just leave this as is for now. I've opened https://github.com/micheles/decorator/issues/164 upstream. Thanks Evgeni!

Reply:
Okay, not a typo - there are 2 stages of `%` replacement, so we want to escape a `%` in the first stage so that it's there for the second stage.